crates/runtime/src/tools/mcp/catalog.rs

```diff
@@ -41,12 +42,21 @@ pub(crate) struct McpToolCatalog {
 
     /// User defined name & description, not from underlying MCP.
     name: String,
-    heartbeat_task: tokio::task::JoinHandle<()>,
+
+    // Must be `Option` to allow taking a mutable reference to the `heartbeat_task` field in the `Drop` implementation.
+    heartbeat_task: Option<CancellableTaskHandle>,
 }
 
 impl Drop for McpToolCatalog {
     fn drop(&mut self) {
-        self.heartbeat_task.abort();
+        if let Some(heartbeat) = self.heartbeat_task.take() {
+            tokio::spawn(heartbeat.cancel(Duration::from_secs(15)));
```

Comment:
This will panic if the tokio runtime is shutdown - which there is a good chance it might be when this is called in a shutdown routine.

Ideally we don't need to do any async work here - is there a way we can just send a cancel signal without waiting for the response?

Reply:
Created `cancel_without_wait`.